client/src/components/ServicesFilter/ServicesFilter.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {listItemSectionBuilder, MultiSelectCheckbox, TableToolbar} from 'wix-style-react';
+import {translate} from '../../utils';
+import {isEqual} from 'lodash';
+
+export default class ServicesFilter extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedServices: props.filters.services
+        };
+    }
+
+    _getOptionsList = (options) => {
+        const {servicesGroups} = this.props;
+        for (const serviceGroup in servicesGroups) {
+            if (servicesGroups[serviceGroup].length) {
+                options.push(listItemSectionBuilder({title: translate(`ServicesFilter.${serviceGroup.toLowerCase()}`)}));
+                options.push(...servicesGroups[serviceGroup]);
+            }
+        }
+    };
+
+    _onServiceOptionSelect = (optionScheduleId) => {
+        this.setState({selectedServices: [...this.state.selectedServices || [], optionScheduleId]});
+    };
+
+    _onServiceOptionDeselect = (optionScheduleId) => {
+        this.setState({selectedServices: this.state.selectedServices.filter((selectedService) => selectedService !== optionScheduleId)});
+    };
+
+    _onServiceFilterClose = () => {
+        const {onFilterChanged, filters} = this.props;
+        if (!isEqual(filters.services, this.state.selectedServices)) {
```

Comment:
if i remember correctly, lodash.isEqual is also consider the order.
so if you were to compare [1, 4, 3] with [3, 4, 1] you would get false result.
this can happen if you unmark some values and then re-mark them in a different order.

think about what can you do to prevent that.